cmd/kyma/alpha/create/module/module.go

```diff
@@ -32,64 +38,102 @@ func NewCmd(o *Options) *cobra.Command {
+- internal:     a bool, optional, default=false, determines whether the ModuleTemplate should have the internal flag or not
+- beta:         a bool, optional, default=false, determines whether the ModuleTemplate should have the beta flag or not
+- labels:       a map with string keys and values, optional, additional labels for the generated ModuleTemplate
+- annotations:  a map with string keys and values, optional, additional annotations for the generated ModuleTemplate
+
+The "manifest" and "defaultCR" paths are resolved against the module's directory, as configured with "--path" flag.
+The "manifest" file contains all the module's resources in a single, multi-document YAML file. These resources will be created in the Kyma cluster when the module is activated.
+The "defaultCR" file contain a default Custom Resource for the module that will be installed along with the module.
+It is additionally schema-validated against the Custom Resource Definition for it's Group/Kind that must exists is the set of resources in the "manifest" file.
+
+### Kubebuilder mode configuration
+The Kubebuilder mode is DEPRECATED.
+The Kubebuilder mode is configured automatically if the "--module-config-file" flag is not provided.
+
+In this mode you have to explicitly provide the module name and version using "--name" and "--version" flags, respectively.
+Some defaults, like the module manifest file location and the default CR file location are then resolved automatically, but you can override these with available flags.
+
+### Modules as OCI artifacts
+Modules are built and distributed as OCI artifacts. 
+This command creates a component descriptor in the configured descriptor path (./mod as a default) and packages all the contents on the provided path as an OCI artifact.
+The internal structure of the artifact conforms to the [Open Component Model](https://ocm.software/) scheme version 3.
+
+If you configured the "--registry" flag, the created module is validated and pushed to the configured registry.
+During the validation the default CR resource, if defined, is validated against a corresponding CustomResourceDefinition.
+You can also trigger an on-demand default CR validation with "--validateCR=true", in case you don't push to the registry.
+
+#### Name Mapping Modes
+To push the artifact into some registries, for example, the central docker.io registry, you have to change the OCM Component Name Mapping with the following flag: "--name-mapping=sha256-digest". This is necessary because the registry does not accept artifact URLs with more than two path segments, and such URLs are generated with the default name mapping: "urlPath". In the case of the "sha256-digest" mapping, the artifact URL contains just a sha256 digest of the full Component Name and fits the path length restrictions. The downside of the "sha256-mapping" is that the module name is no longer visible in the artifact URL, as it contains the sha256 digest of the defined name.
```

Comment:
```suggestion
To push the artifact into some registries, for example, the central docker.io registry, you have to change the OCM Component Name Mapping with the following flag: `--name-mapping=sha256-digest`. This is necessary because the registry does not accept artifact URLs with more than two path segments, and such URLs are generated with the default name mapping: **urlPath**. In the case of the `sha256-digest` mapping, the artifact URL contains just a sha256 digest of the full Component Name and fits the path length restrictions. The downside of the `sha256-mapping` is that the module name is no longer visible in the artifact URL, as it contains the sha256 digest of the defined name.
```